edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeClientWebSocket.cs

```diff
@@ -26,13 +30,28 @@ public static async Task<IClientWebSocket> Connect(Uri url, string authToken, Ca
 
         public WebSocketState State => this.clientWebSocket.State;
 
-        public Task CloseAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken)
-            => this.clientWebSocket.CloseAsync(closeStatus, statusDescription, cancellationToken);
+        public async Task CloseAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken)
+        {
+            using (await this.clientWebSocketLock.LockAsync(cancellationToken))
```

Comment:
Is this cancellation token always set to something non-none?

Reply:
Yes - for the streaming case, we have a real cancellation token, which is cancelled if the client closes, etc.